### PR TITLE
docs: Interactive SizeEffect examples in doc.

### LINKED ISSUE #1952: Interactive docs: `SizeEffect`
# Interactive doc for the `SizeEffect`s

The two `SizeEffect`s should have small examples that show how the effect is applied and how it works. This could for example be just clicking somewhere in the game and resizing a geometrical shape.

https://docs.flame-engine.org/1.3.0/flame/effects.html#sizeeffect-by
https://docs.flame-engine.org/1.3.0/flame/effects.html#sizeeffect-to

# What are interactive docs?

On our documentation site has the possibility to run inline interactive examples and we would like every visual feature to have a **small**, pretty and easily understandable example attached to it. Some current examples are the [router](https://docs.flame-engine.org/1.3.0/flame/router.html) and the [decorators](https://docs.flame-engine.org/1.3.0/flame/rendering/decorators.html).
Some inspiration can also be found on our [examples page](https://examples.flame-engine.org) where we have examples for most features.

# Good to know

Make sure that you comment on the issue that you will be working on this before starting to program.

To set up the Flame environment you need to follow the instructions in our [contributing.md](https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md#environment-setup) doc, if you have any problems setting up your environment (or other issues of course) feel free to ask for help in the Flame channel on our [Discord server](https://discord.com/invite/pxrBmy4).

Since this is a documentation change, you need to be able to build the docs. Instructions for building the documentation site locally can be found here: https://docs.flame-engine.org/main/development/documentation.html#building-documentation-locally

Make sure to include a video of the example when you upload the PR so that it will be easier for us to review it. (You can attach videos directly in the PR, so no need to upload them somewhere else)

**Happy hacking!**

### PR DESCRIPTION
**Pull request by idlehart**
# Description
Doc examples for SizeEffect.to and SizeEffect.by.
I used the Ember because the Flower gave script errors when trying to size.

## Checklist

- [x] The title of my PR starts with a [Conventional Commit] prefix (`fix:`, `feat:`, `docs:` etc).
- [x] I have followed the [Contributor Guide] when preparing my PR.
- [ ] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [x] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?

- [ ] Yes, this PR is a breaking change.
- [x] No, this PR is not a breaking change.

## Related Issues

Closes #1952

[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Conventional Commit]: https://conventionalcommits.org

### CODE REVIEW

**Comment by munsterlander** [comment]
So excited to see Ember used!

File: doc/flame/examples/lib/main.dart [hunk 1]
```diff
@@ -50,6 +52,8 @@ void main() {
     'router': RouterGame.new,
     'scale_by_effect': ScaleByEffectGame.new,
     'scale_to_effect': ScaleToEffectGame.new,
+    'size_by_effect': SizeByEffectGame.new,
+    'size_to_effect': SizeToEffectGame.new,
```
Comment (spydon) [inline doc/flame/examples/lib/main.dart:56]: These shouldn't be indented

**Review by spydon — APPROVED** [review]
Strange that it didn't work with the flower, but looks good except for that!
Had another look at `Flower` and it doesn't use `size` when it is rendering so the `SizeEffect` wouldn't have any visual effect on it, I wonder why it is getting that error from Flutter though...

**Comment by munsterlander** [comment]
> Strange that it didn't work with the flower, but looks good except for that!

Flower doesn't work because sizes are specified in the class itself.  This is the same for opacity and was one of the reasons for that whole thing.

File: doc/flame/examples/lib/main.dart (hunk 1 above)
Comment (idlehart) [reply doc/flame/examples/lib/main.dart:56]: Not sure what you mean here, don't they follow the other routes' pattern?

Comment (spydon) [reply doc/flame/examples/lib/main.dart:56]: It looked super weird when reviewing it on the phone, looks completely fine in reality!

**Comment by spydon** [comment]
Now I know why this worked on `Ember`, but not on `Flower`, it's due to `SpriteAnimationComponent` having the `SizeProvider` and not `PositionComponent`. And if you would have run this code in debug mode you'd have got an assert error instead of the weird error that you got.

https://github.com/flame-engine/flame/issues/2087

**Pull request merged**